### PR TITLE
[CARBONDATA-488][SPARK2]add InsertInto feature for spark2

### PR DESCRIPTION
**Pull request by QiangCai**
1. add InsertInto feature for spark2

2. optimize CarbonExample to use relation path
    And use InsertInto to load data

Link:
[CARBONDATA-488](https://issues.apache.org/jira/browse/CARBONDATA-488)

### CODE REVIEW

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonExample.scala [hunk 1]
```diff
@@ -17,79 +17,97 @@
...
-    // to run the example, plz change this path to your local machine path
-    val rootPath = "/home/david/Documents/incubator-carbondata"
+    val rootPath = new File(this.getClass.getResource("/").getPath
+        + "../../../..").getCanonicalPath
+    val storeLocation = s"$rootPath/examples/spark2/target/store"
+    val warehouse = s"$rootPath/examples/spark2/target/warehouse"
+    val metastoredb = s"$rootPath/examples/spark2/target/metastore_db"
+
+    // clean data folder
+    if (true) {
```
Comment (jackylk) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonExample.scala]: hard coded true?

Comment (QiangCai) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonExample.scala]: this is easy to change to false, if developer want to keep data.

**Comment by jackylk** [comment]
LGTM

**Comment by jackylk** [comment]
CI
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/735/

**Pull request closed without merging**